File: app/src/main/java/me/yuu/sample/MainActivity.java
package me.yuu.sample;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import me.yuu.liteadapter.core.LiteAdapter;
import me.yuu.liteadapter.core.ViewHolder;
import me.yuu.liteadapter.core.ViewInjector;

/**
 * @author yu
 */
public class MainActivity extends AppCompatActivity {

    private final List<SampleEntity> data = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_list);

        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        LiteAdapter<SampleEntity> adapter = new LiteAdapter.Builder<SampleEntity>(this)
                .register(new ViewInjector<SampleEntity>(R.layout.item_main) {
                    @Override
                    public void bindData(ViewHolder holder, SampleEntity item, int position) {
                        holder.setText(R.id.tvDesc, item.getName());
                    }
                })
                .itemClickListener(new LiteAdapter.OnItemClickListener() {
                    @Override
                    public void onItemClick(int position, Object item) {
                        startActivity(new Intent(MainActivity.this,
                                ((SampleEntity) item).getTarget()));
                    }
                })
                .create()
                .attachTo(recyclerView);

        adapter.setNewData(data);
    }

    {
        data.add(new SampleEntity("Header & Footer", HeaderAndFooterActivity.class));
        data.add(new SampleEntity("Empty view & Auto load more", EmptyAndLoadMoreActivity.class));
    }
}

File: app/src/main/java/me/yuu/sample/ui/MainActivity.kt
package me.yuu.sample.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.layout_list.*
import me.yuu.liteadapter.ext.buildAdapter
import me.yuu.sample.R
import me.yuu.sample.entity.SampleEntity
import java.util.*

/**
 * @author yu
 */
class MainActivity : AppCompatActivity() {

    private val data = ArrayList<SampleEntity>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_list)

        val adapter = buildAdapter<SampleEntity>(this) {
            register(R.layout.item_main) { holder, item, _ ->
                holder.setText(R.id.tvDesc, item.name)
            }
            itemClickListener { _, item ->
                startActivity(Intent(this@MainActivity, (item as SampleEntity).target))
            }
        }

        recyclerView.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            this.adapter = adapter
        }

        adapter.updateData(data)
    }

    init {
        data.add(SampleEntity("Header & Footer", HeaderAndFooterActivity::class.java))
        data.add(SampleEntity("Empty view & Auto load more", EmptyAndLoadMoreActivity::class.java))
        data.add(SampleEntity("DataBinding", DataBindingSampleActivity::class.java))
        data.add(SampleEntity("RadioActivity", RadioActivity::class.java))
    }
}
